packages/flutter/lib/src/cupertino/context_menu.dart

```diff
@@ -605,6 +615,14 @@ class _CupertinoContextMenuState extends State<CupertinoContextMenu> with Ticker
       _childHidden = true;
     });
 
+    Future<void>.delayed(const Duration(milliseconds: 350), () {
+      if (_openController.isAnimating &&
+          _openController.status != AnimationStatus.reverse &&
+          widget.enableHapticFeedback) {
+        HapticFeedback.heavyImpact();
+      }
+    });
```

Comment:
Typically we try to avoid using timeouts as much as possible outside of tests and error code in order to avoid race conditions. I'm wondering if the same thing can be solved by adding a listener to the animation controller that will trigger the feedback once after it passes a certain value in the forward direction?

Reply:
Yea that makes sense. Pushed a new commit now with this fix :)